src/session/session_api.c

```diff
@@ -303,6 +303,19 @@ __session_close(WT_SESSION *wt_session, const char *config)
     SESSION_API_CALL_PREPARE_ALLOWED(session, close, config, cfg);
     WT_UNUSED(cfg);
 
+    /*
+     * Ideally tracking would be after the close call, but that gets complex. The api tracking
+     * macros don't balance the count since they rely on a valid session handle. Once the session
+     * handle is gone, we can't rely on the connection remaining valid. The internal session close
+     * API is called internally a lot, so it isn't appealing to split that out to improve this
+     * tracking. Let's live with the imprecision.
+     */
+    if (!WT_SESSION_IS_DEFAULT(session) && session->api_call_counter == 1) {
+        if (F_ISSET(session, WT_SESSION_INTERNAL))
+            (void)__wt_atomic_add64(&S2C(session)->api_count_int_out, 1);
+        else
+            (void)__wt_atomic_add64(&S2C(session)->api_count_out, 1);
+    }
```

Comment:
Could do something like:
```
     WT_CONNECTION_IMPL *track_conn;
     bool internal;
....
     internal = F_ISSET(session, WT_SESSION_INTERNAL);
     if (!WT_SESSION_IS_DEFAULT(session) && session->api_call_counter == 1)
        track_conn = S2C(session);
     else
        track_conn = NULL;

      WT_TRET(__wt_session_close_internal(session));
      session = NULL;
      if (track_conn) {
         if (internal)
              wt_atomic_add(...);
         else
              wt_atomic_add(...);
   }
```

Reply:
Thanks. I have been trying to figure out how accurate this should be. Obviously it would be best if it was entirely accurate, but I also don't think we need to worry too much about cases like this.

From my point of view the internal tracking was really a bit of free information - since I initially didn't distinguish and was surprised that concurrent activity was sometimes higher than workload activity.

The likelihood that a meaningful amount of time is spent inside WiredTiger in session close seems low enough that I'm not concerned about getting it right.

I left this big comment here primarily so that a subsequent API entry/exit macro tracking change would think about getting this right (if possible).

Let me know if you think it's worth getting this right - I'll be happy to take your suggested change (thanks for taking the time).

Reply:
I don't think it's very important, as you say, the close call should be pretty fast.  "Fixing" it does make the code more complex, so I'm heppy with the original PR.